www/js/lib/zimArchive.js

```diff
@@ -94,13 +94,13 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8', 'uiUtil'],
                 that._searchArchiveParts(storage, path.slice(0, -2)).then(function(fileArray) {
                     createZimfile(fileArray);
                 }, function(error) {
-                    uiUtil.systemAlert("Error reading files in split archive " + path + ": " + error, "Error reading archive files");
+                    callbackError("Error reading files in split archive " + path + ": " + error, "Error reading archive files");
```

Comment:
Here (& below) the error callback is being called in place of the second argument of `.then` block. Should we call it in a separate catch block instead? If this can work then it can achieve a flatter code I suppose.

Reply:
Yes, that seems to be using an old-fashioned Promise style. You could test using a catch block here and below instead. And if so, you might be able to report the error in the catch block of the calling function instead of having to have a callbackError. But I haven't looked at the code that calls this.